src/modules/NtpClient.rb

```diff
@@ -1062,22 +1082,37 @@ def new_write_progress
       )
     end
 
+    def update_cfa_record(record)
+      cfa_record = record["cfa_record"]
+      cfa_record.value = record["address"]
+      cfa_record.raw_options = record["options"]
+      cfa_record.comment = record["comment"]
+    end
+
     # Write current /etc/ntp.conf with @ntp_records
     # @return [Boolean] true on success
     def write_ntp_conf
-      conf = SCR.Read(path(".etc.ntp_conf.all"))
-
-      return false if conf.nil?
-
-      conf["value"] = records_for_write
+      records_for_write.each do |record|
+        unless record["cfa_record"]
+          record["cfa_record"] = CFA::NtpConf::Record.record_class(record["type"]).new
+          @ntp_conf.records << record["cfa_record"]
+        end
+        update_cfa_record(record)
+      end
 
-      return false if !SCR.Write(path(".etc.ntp_conf.all"), conf)
+      @deleted_records.each do |cfa_record|
+        @ntp_conf.records.delete(cfa_record)
+      end
```

Comment:
you can implement it with delete_if `@ntp_conf.records.delete_if { |r| @deleted_records.include?(r) }`